flux-syntax/src/surface.rs

```diff
@@ -368,3 +372,13 @@ impl fmt::Debug for BinOp {
         }
     }
 }
+
+pub fn async_return_ty(ty: Ty) -> Ty {
```

Comment:
I would carry the asyncness and deal with this later (maybe during `conv`) but I guess this is fine for now.